modules/platforms/cpp/core/include/ignite/impl/cache/event/cache_entry_event_filter_holder.h

```diff
@@ -176,6 +178,60 @@ namespace ignite
                         return 0;
                     }
                 };
+
+                class JavaCacheEntryEventFilterHolder : public CacheEntryEventFilterHolderBase
+                {
+                public:
+                    /**
+                     * Default constructor.
+                     */
+                    JavaCacheEntryEventFilterHolder()
+                    {
+                        // No-op.
+                    }
+
+                    /**
+                     * Constructor.
+                     */
+                    JavaCacheEntryEventFilterHolder(const ignite::cache::event::JavaCacheEntryEventFilter& filter) :
+                        filter(filter)
+                    {
+                        // No-op.
+                    }
+
+                    /**
+                     * Destructor.
+                     */
+                    virtual ~JavaCacheEntryEventFilterHolder()
+                    {
+                        // No-op.
+                    }
+
+                    /**
+                     * Process input.
+                     *
+                     * @param writer Writer.
+                     */
+                    virtual void Write(binary::BinaryWriterImpl& writer)
+                    {
+                        writer.WriteBool(true);
+                        writer.WriteObject<impl::PlatformJavaObjectFactoryProxy>(*filter.factory.Get());
+                    }
+
+                    /**
+                     * Get filter pointer.
+                     *
+                     * @return Filter.
+                     */
+                    virtual CacheEntryEventFilterBase* GetFilter()
+                    {
+                        return 0;
```

Comment:
Perhaps add a comment explaining why this is `0`?
And perhaps general info what is the difference between holding `JavaCacheEntryEventFilter` instance vs `CacheEntryEventFilterBase`?

Reply:
Added comment. Basically, `JavaCacheEntryEventFilter` is interface that provides us with ability to read remote filter passed to us by Java layer and execute it on specific data. This is needed when we want to use remote filters written in C++ on remote nodes. `JavaCacheEntryEventFilter` on the other hand is just a name of a Java class with the set of parameters, that should be called on Java side and should never be passed to C++ layer.